### ISSUE TITLE
NPE when comparing RealmConfigurations with no Rx factory specified

### ISSUE THREAD

**Issue by zakaprov** [realm/realm-java#2475]
Hey there! First of all - big cheers to the Realm team, really! I've been using Realm for a year now and have always loved the product. Looking forward to seeing future updates of this project!

#### Goal

Inject an instance of Realm via Dagger. The values for the RealmConfiguration object are reused, creating an instance using `Realm#getInstance(realmConfiguration)`.

#### Expected Results

Realm gets instantiated and injected properly in all activities, not just the first one using it.

#### Actual Results

```
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
at android.app.ActivityThread.-wrap11(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5417)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
at io.realm.RealmConfiguration.equals(RealmConfiguration.java:186)
at io.realm.RealmCache.validateConfiguration(RealmCache.java:217)
at io.realm.RealmCache.createRealmOrGetFromCache(RealmCache.java:103)
at io.realm.Realm.getInstance(Realm.java:213)
```

Injecting for the first time works fine, for any other attempt - it fails. As I followed the debugger down Realm's code I noticed that this happens on cache hit when obtaining a Realm instance.
`RealmCache#validateConfiguration` attempts to compare the new `RealmConfiguration` with the one from cache.
`RealmConfiguration#equals` throws an NPE when comparing `RxObservableFactory`, as it's not provided in any of these configs.

#### Debugging

This can be avoided by adding `.rxFactory(new RealmObservableFactory())` call to the `RealmConfiguration` builder.

The Realm docs state this:

> If no RxObservableFactory is defined, Realm defaults to RealmObservableFactory which is a class provided by Realm that supports RxJava <= 1.1.*.

I guess it should be possible to not specify the Rx factory, since not every project needs to depend on RxJava.

#### Version of Realm and tooling
Realm version(s): 0.88.0 (Gradle plugin)

Android Studio version: 1.5.1

Which Android version and device: This specific case was tested on a Genymotion emulated Nexus 6 running 6.0.

**Comment by beeender** [comment]
Thanks a lot for the bug report! Luckily it should be a same issue with #2416 and fixed by #2419 .

The latest release includes this fix, so give it a try? :beers:

**Comment by zakaprov** [comment]
Great! Thanks a lot for your response and sorry about the duplicate. Closing the issue now.
